src/routes/(console)/project-[region]-[project]/sites/site-[site]/domains/add-domain/+page.svelte

```diff
@@ -50,27 +52,41 @@
     });
 
     async function addDomain() {
+        let domain = data.domains?.domains.find((d) => d.domain === domainName);
+
+        if (!domain && isCloud) {
+            try {
+                domain = await sdk.forConsole.domains.create($project.teamId, domainName);
+            } catch (error) {
+                addNotification({
+                    type: 'error',
+                    message: error.message
+                });
+
+                return;
+            }
+        }
+
         try {
             let rule: Models.ProxyRule;
             if (behaviour === 'BRANCH') {
                 rule = await sdk
                     .forProject(page.params.region, page.params.project)
-                    .proxy.createSiteRule(domainName, page.params.site, branch);
+                    .proxy.createSiteRule(domain.domain, page.params.site, branch);
             } else if (behaviour === 'REDIRECT') {
-                const sc = Object.values(StatusCode).find((code) => parseInt(code) === statusCode);
                 rule = await sdk
                     .forProject(page.params.region, page.params.project)
-                    .proxy.createRedirectRule(domainName, $protocol + redirect, sc);
+                    .proxy.createRedirectRule(domain.domain, $protocol + redirect, statusCode);
             } else if (behaviour === 'ACTIVE') {
                 rule = await sdk
                     .forProject(page.params.region, page.params.project)
-                    .proxy.createSiteRule(domainName, page.params.site);
+                    .proxy.createSiteRule(domain.domain, page.params.site);
             }
             if (rule?.status === 'verified') {
                 await goto(routeBase);
                 await invalidate(Dependencies.SITES_DOMAINS);
             } else {
-                await goto(`${routeBase}/add-domain/verify-${domainName}?rule=${rule.$id}`);
+                await goto(`${routeBase}/add-domain/verify-${domain.domain}?rule=${rule.$id}`);
```

Comment:
`domain.domain` won't exist on selfhosted... I think we need to use `domainName` here